echopype/convert/api.py

```diff
@@ -427,28 +429,43 @@ def open_raw(
         sonar_model=sonar_model,
         params=_set_convert_params(convert_params),
     )
-    # Set up echodata object
-    echodata = EchoData(source_file=file_chk, xml_path=xml_chk, sonar_model=sonar_model)
+
+    # Setup tree dictionary
+    tree_dict = {}
     # Top-level date_created varies depending on sonar model
     if sonar_model in ["EK60", "ES70", "EK80", "ES80", "EA640"]:
-        echodata.top = setgrouper.set_toplevel(
-            sonar_model=sonar_model, date_created=parser.config_datagram["timestamp"]
+        tree_dict["Top-level"] = setgrouper.set_toplevel(
+            sonar_model=sonar_model,
+            date_created=parser.config_datagram["timestamp"],
         )
     else:
-        echodata.top = setgrouper.set_toplevel(
+        tree_dict["Top-level"] = setgrouper.set_toplevel(
             sonar_model=sonar_model, date_created=parser.ping_time[0]
         )
-    echodata.environment = setgrouper.set_env()
-    echodata.platform = setgrouper.set_platform()
+    tree_dict["Environment"] = setgrouper.set_env()
+    tree_dict["Platform"] = setgrouper.set_platform()
     if sonar_model in ["EK60", "ES70", "EK80", "ES80", "EA640"]:
-        echodata.nmea = setgrouper.set_nmea()
-    echodata.provenance = setgrouper.set_provenance()
-    echodata.sonar = setgrouper.set_sonar()
-    # Beam_group2 group only exist if EK80 has both complex and power/angle data
-    if sonar_model in ["EK80", "ES80", "EA640"]:
-        echodata.beam, echodata.beam_power = setgrouper.set_beam()
-    else:
-        echodata.beam = setgrouper.set_beam()
-    echodata.vendor = setgrouper.set_vendor()
+        tree_dict["Platform/NMEA"] = setgrouper.set_nmea()
+    tree_dict["Provenance"] = setgrouper.set_provenance()
+    tree_dict["Sonar"] = setgrouper.set_sonar()
+
+    # Set multi beam groups
+    beam_groups = setgrouper.set_beam()
+    if isinstance(beam_groups, xr.Dataset):
+        # if it's a single dataset like the ek60,
+        # make into list
+        beam_groups = [beam_groups]
+
+    for idx, beam_group in enumerate(beam_groups, start=1):
+        tree_dict[f"Sonar/Beam_group{idx}"] = beam_group
+
+    tree_dict["Vendor"] = setgrouper.set_vendor()
+
+    # Create tree and echodata
+    # TODO: make the creation of tree dynamically generated from yaml
```

Comment:
Just curious about this TODO. Is your idea here to replace the above code with a dynamic approach?

Reply:
Yepp, this is from a source of truth somewhere? I'm not sure to be honest.

Reply:
Sounds good!